pkg/kubelet/userns/userns_manager_test.go

```diff
@@ -171,3 +183,179 @@ func TestUserNsManagerParseUserNsFile(t *testing.T) {
+			}
+		})
+	}
+}
+
+func TestCleanupOrphanedPodUsernsAllocations(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.UserNamespacesSupport, true)()
+
+	cases := []struct {
+		name                 string
+		runningPods          []*kubecontainer.Pod
+		pods                 []*v1.Pod
+		listPods             []types.UID /* pods to list */
+		podSetBeforeCleanup  []types.UID /* pods to record before cleanup */
+		podSetAfterCleanup   []types.UID /* pods set expected after cleanup */
+		podUnsetAfterCleanup []types.UID /* pods set expected after cleanup */
+	}{
+		{
+			name:     "no stale pods",
+			listPods: []types.UID{"pod-1", "pod-2"},
+		},
+		{
+			name:                 "no stale pods set",
+			podSetBeforeCleanup:  []types.UID{"pod-1", "pod-2"},
+			listPods:             []types.UID{"pod-1", "pod-2"},
+			podUnsetAfterCleanup: []types.UID{"pod-1", "pod-2"},
+		},
+		{
+			name:                 "one running pod",
+			listPods:             []types.UID{"pod-1", "pod-2"},
+			podSetBeforeCleanup:  []types.UID{"pod-1", "pod-2"},
+			runningPods:          []*kubecontainer.Pod{{ID: "pod-1"}},
+			podSetAfterCleanup:   []types.UID{"pod-1"},
+			podUnsetAfterCleanup: []types.UID{"pod-2"},
+		},
+		{
+			name:                 "pod set before cleanup but not listed ==> unset",
+			podSetBeforeCleanup:  []types.UID{"pod-1", "pod-2"},
+			runningPods:          []*kubecontainer.Pod{{ID: "pod-1"}},
+			podUnsetAfterCleanup: []types.UID{"pod-1", "pod-2"},
+		},
+		{
+			name:                 "one pod",
+			listPods:             []types.UID{"pod-1", "pod-2"},
+			podSetBeforeCleanup:  []types.UID{"pod-1", "pod-2"},
+			pods:                 []*v1.Pod{{ObjectMeta: metav1.ObjectMeta{UID: "pod-1"}}},
+			podSetAfterCleanup:   []types.UID{"pod-1"},
+			podUnsetAfterCleanup: []types.UID{"pod-2"},
+		},
+		{
+			name:                 "no listed pods ==> all unset",
+			podSetBeforeCleanup:  []types.UID{"pod-1", "pod-2"},
+			podUnsetAfterCleanup: []types.UID{"pod-1", "pod-2"},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			testUserNsPodsManager := &testUserNsPodsManager{
+				podList: tc.listPods,
+			}
+			m, err := MakeUserNsManager(testUserNsPodsManager)
+			require.NoError(t, err)
+
+			// Record the userns range as used
+			for i, pod := range tc.podSetBeforeCleanup {
+				err := m.record(pod, uint32((i+1)*65536), 65536)
+				require.NoError(t, err)
+			}
+
+			err = m.CleanupOrphanedPodUsernsAllocations(tc.pods, tc.runningPods)
+			require.NoError(t, err)
+
+			for _, pod := range tc.podSetAfterCleanup {
+				ok := m.podAllocated(pod)
+				assert.True(t, ok, "pod %q should be allocated", pod)
+			}
+
+			for _, pod := range tc.podUnsetAfterCleanup {
+				ok := m.podAllocated(pod)
+				assert.False(t, ok, "pod %q should not be allocated", pod)
+			}
+		})
+	}
+}
+
+func TestAllocateOneMaxPods(t *testing.T) {
```

Comment:
nit:  Maybe call it TestMaxPodsAllocation

Reply:
or TestAllocateMaxPods like the next test.

Reply:
Fixed, thanks!